### ISSUE TITLE
fix: Loss stuck at 0.0000 and scheduler.step(epoch) deprecation in train_dit.py

### ISSUE THREAD

**Issue by d-oit** [d-oit/tiny-cats-model#65]
## Summary

Two persistent bugs observed in the Modal GPU training run [`ap-PoaPbeZ3mXeSn6cL2jZ3vV`](https://modal.com/apps/d-oit/main/ap-PoaPbeZ3mXeSn6cL2jZ3vV) on 2026-05-01.

---

## Bug 1: Loss always reports `0.0000`

**Symptom:** Every step from 300–2200 logs `Loss: 0.0000`, which is a logging bug — not evidence of good training.

**Root cause:** The log format string uses `:.4f` which silently rounds tiny values (e.g. `0.00003`) to `0.0000`.

**Fix in `train_dit.py` log interval block:**
```python
# BAD
logger.info(f"Loss: {avg_loss:.4f}")

# GOOD — scientific notation reveals actual value
logger.info(f"Loss: {avg_loss:.6e}")
```

Also verify `loss.item()` is called before accumulating into `running_loss`:
```python
running_loss += loss.item()  # .item() must be called
```

---

## Bug 2: `scheduler.step(epoch)` deprecation warning

**Symptom at step 2,000:**
```
UserWarning: The epoch parameter in `scheduler.step()` was not necessary and is being deprecated...
```

**Fix:** Replace any `scheduler.step(current_step)` calls with a `LambdaLR` scheduler:
```python
import math

def lr_lambda(current_step: int):
    if current_step < warmup_steps:
        return float(current_step) / float(max(1, warmup_steps))
    progress = (current_step - warmup_steps) / max(1, total_steps - warmup_steps)
    return max(0.0, 0.5 * (1.0 + math.cos(math.pi * progress)))

scheduler = torch.optim.lr_scheduler.LambdaLR(optimizer, lr_lambda)

# In training loop — no argument:
scheduler.step()
```

---

## Environment
- GPU: NVIDIA A10G (Modal)
- Python 3.12 / PyTorch with AMP
- Model: TinyDiT 33M params, 128×128
- Steps: 100,000 | Batch: 512 | LR: 5e-5 | Warmup: 2,000
